### PR TITLE
Allow deleting gist

### PR DESCRIPTION
**Pull request by hkalexling**
For issue #307

### CODE REVIEW

File: OctoKit/OCTClient+Gists.h [hunk 1]
```diff
@@ -45,4 +45,12 @@
 // is not `authenticated`, the signal will error immediately.
 - (RACSignal *)createGistWithEdit:(OCTGistEdit *)edit;
 
+// Delete a gist
+//
+// gsit - The gist to delete. This must not be nill.
```
Comment (joshaber) [inline OctoKit/OCTClient+Gists.h]: Lil type here: s/gsit/gist

File: OctoKit/OCTClient+Gists.m [hunk 2]
```diff
@@ -52,4 +52,14 @@ - (RACSignal *)createGistWithEdit:(OCTGistEdit *)edit {
 	return [[self enqueueRequest:request resultClass:OCTGist.class] oct_parsedResults];
 }
 
+- (RACSignal *)deleteGist:(OCTGist *)gist {
+	NSParameterAssert(gist != nil);
+	
+	if (!self.authenticated)
```
Comment (joshaber) [inline OctoKit/OCTClient+Gists.m]: Style: this `if` should have curly braces around its body.

**Comment by joshaber** [comment]
Thanks! Just a couple comments.

**Comment by hkalexling** [comment]
@joshaber fixed! Please have a look :)

**Comment by joshaber** [comment]
Looks good, thanks!

**Pull request merged**